device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsCbsReceiverLinkHandler.java

```diff
@@ -70,7 +70,7 @@ private void handleCBSResponseMessage(Receiver receiver)
             if (amqpsMessage.getApplicationProperties() != null && amqpsMessage.getProperties() != null)
             {
                 Properties properties = amqpsMessage.getProperties();
-                Object correlationId = properties.getCorrelationId();
+                UUID correlationId = (UUID) properties.getCorrelationId();
```

Comment:
Proton's message's correlationId's are of type Object, but we only ever assign them to be instances of UUIDs, so this is a safe cast to do